gobblin-api/src/main/java/gobblin/configuration/WorkUnitState.java

```diff
@@ -287,21 +311,22 @@ public boolean equals(Object object) {
     }
 
     WorkUnitState other = (WorkUnitState) object;
-    return ((this.workunit == null && other.workunit == null)
-        || (this.workunit != null && this.workunit.equals(other.workunit))) && super.equals(other);
+    return ((this.workUnit == null && other.workUnit == null)
```

Comment:
Should this method and `hashCode()` be updated to include the `jobState`. Alternatively, we could just use lombok's `@EqualsAndHashCode`